autonomi/src/client/error.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 MaidSafe.net limited.
```

Comment:
I don't think we should put all errors in one module. The idea is for futures to be very specific, so we might end up with way more error types. We've already become lazy recently by just piling error variants up into these structs.

If you look at `std`, you'll see all errors also with their respective functions or modules. It keeps it much easier to edit and find the related error types.

Reply:
Oh okay, fair point